apps/site/src/pages/account/general-panel/general-panel.tsx

```diff
@@ -0,0 +1,277 @@
+            <>
+              Information that may appear on your public Block Protocol profile
+              page, visible <Link href={`/@${user.shortname}`}>here</Link>.
+            </>
+          }
+        >
+          <Stack direction={isMobile ? "column" : "row"} gap={4.5}>
+            <Stack flex={1} gap={3}>
+              <SaveOnBlurTextField
+                fullWidth
+                placeholder="e.g. John"
+                label="First or preferred name"
+                helperText="Your name is visible on your public profile"
+                value={user.preferredName}
+                onChange={(event) => {
+                  setUser({
+                    ...user,
+                    preferredName: event.target.value.trim(),
+                  });
+                }}
+                onSave={async (preferredName) => {
+                  /** @todo handle error state */
+                  await apiClient.updateUserPreferredName({ preferredName });
+                }}
+              />
+              <TextField
+                disabled
+                fullWidth
+                label="Username (cannot be changed)"
+                value={`@${user.shortname}`}
+                helperText="Your username is your public handle and cannot be changed"
+              />
+            </Stack>
+
+            <Box flex={1}>
+              <Typography
+                sx={{ color: "gray.80", fontSize: 15, fontWeight: 500 }}
+              >
+                Avatar
+              </Typography>
+              <Stack direction={isMobile ? "column" : "row"}>
+                <AvatarWithOverlay
+                  src={user.userAvatarUrl}
+                  mode={
+                    isDragging
+                      ? "drop"
+                      : displayRemoveAvatarConfirmation
+                      ? "danger"
+                      : "idle"
+                  }
+                  onDragOver={handleDrag}
+                  onDrop={handleDrop}
+                />
+
+                <Stack
+                  sx={{
+                    alignItems: "flex-start",
+                    flex: 1,
+                    gap: 1,
+                    alignSelf: isMobile ? "stretch" : "center",
+                  }}
+                >
+                  {isDragging ? (
+                    <AvatarDropzoneInfo isReplacing={!!user.userAvatarUrl} />
+                  ) : displayRemoveAvatarConfirmation ? (
+                    <RemoveAvatarConfirmation
+                      onCancel={() => setDisplayRemoveAvatarConfirmation(false)}
+                      onConfirm={removeAvatar}
+                    />
+                  ) : (
+                    <>
+                      <AvatarButton
+                        onClick={clickOnInput}
+                        startIcon={<FontAwesomeIcon icon={faUpload} />}
+                      >
+                        Upload {hasAvatar ? "new" : "an"} avatar
+                      </AvatarButton>
```

Comment:
There's currently no indication that an upload is in progress and this button is enabled during it (try on slow 3G) – maybe we can disable the button and have it read 'Uploading...' while it's uploading?